ngraph/core/reference/src/runtime/reference/random_uniform.cpp

```diff
@@ -0,0 +1,285 @@
+    return x_float16 - static_cast<float16>(1);
+}
+
+// Helper function for converting uint32 values to double. Sets fractional part of
+// floating double with bits from uint32 values. Resulting value is in interval [0,1).
+double uint32_to_double(uint32_t x1, uint32_t x2) {
+    // float64 is formatted as follows: sign(1 bit) exponent(11 bits) mantissa(52 bits). The value is interpreted
+    // The value is interpreted using following formula:
+    // (-1)^sign * 1, mantissa * 2 ^ (exponent - 1023)
+    // Here we set the following values:
+    // sign = 0
+    // exponent = 1023, for obtaining a zero exponent.
+    // mantissa = 52 right bits from two concatenated uint32 values from random integer generator.
+
+    uint64_t significant = ((static_cast<uint64_t>(x1) & 0xfffffu) << 32) | static_cast<uint64_t>(x2);
+    uint64_t x_uint64 = ((static_cast<uint64_t>(1023) << 52) | significant);
+
+    double x_double;
+    memcpy(&x_double, &x_uint64, sizeof(x_uint64));
+    return x_double - 1.0;
+}
+
+// Helper function for converting uint32 values to bfloat16. Sets fractional part of
+// floating value with bits from uint32 value. Resulting value is in interval [0,1).
+bfloat16 uint32_to_bfloat16(uint32_t x) {
+    // bfloat16 is formatted as follows: sign(1 bit) exponent(8 bits) mantissa(7 bits). The value is interpreted
+    // The value is interpreted using following formula:
+    // (-1)^sign * 1, mantissa * 2 ^ (exponent - 127)
+    // Here we set the following values:
+    // sign = 0
+    // exponent = 127, for obtaining a zero exponent.
+    // mantissa = 7 right bits from generated uint32 random value.
+
+    uint16_t x_uint16 = static_cast<uint16_t>(x);
+    x_uint16 = (static_cast<uint16_t>(127) << 7) | (x_uint16 & 0x7fu);
+
+    bfloat16 x_bfloat16;
+    memcpy(&x_bfloat16, &x_uint16, sizeof(x_uint16));
+    return x_bfloat16 - static_cast<bfloat16>(1);
+}
+
+// Runs Philox algorithm.
+void run_philox(uint64_t key, uint64_t counter, uint64_t n, size_t n_rounds, std::vector<uint32_t>& res) {
+    for (size_t i = 0; i < n_rounds; i++) {
+        calculate_round(key, counter, n);
+        if (i < n_rounds - 1)
+            raise_key(key);
+    }
+    auto res1 = split_high_low(n);
+    auto res2 = split_high_low(counter);
+    res[0] = res1.first;
+    res[1] = res1.second;
+    res[2] = res2.first;
+    res[3] = res2.second;
+}
+
+// Implementation of RandomUniform that uses Philox algorithm as inner random unsigned integer generator.
+void random_uniform(const uint64_t* out_shape,
+                    const char* min_val,
+                    const char* max_val,
+                    char* out,
+                    const Shape& out_shape_shape,
+                    const ngraph::element::Type& elem_type,
+                    uint64_t seed,
+                    uint64_t seed2) {
+    if (seed == 0 && seed2 == 0) {
+        std::srand(std::time(nullptr));
+        seed = std::rand();
+    }
```

Comment:
is this aligned with specification because I didn't find a case with zero seeds there?

Reply:
I though that we can be fully independent from STL random generators. Can plugin teams implement this case?

Reply:
This is aligned with Tensorflow API where zero seeds mead non deterministic sequence. Looks like I need to add this information to specification. 
We discussed this case with plugin teams and they said that they will implement this case as they can use current time as seed initialization.

Reply:
Added comment as discussed offline.